### ISSUE TITLE
Temperature v.s. Base Temperature

### ISSUE THREAD

**Issue by mmderakhshani** [sayakpaul/Supervised-Contrastive-Learning-in-TensorFlow-2#3]
Hi,

Thanks for providing your code. I have read your loss function and I found that there are two temperatures (temperature and base_temperate). could you please tell me what is the difference between these two parameters? I am looking for the effect of base_temperature. 

Thanks.

**Comment by sayakpaul** [comment]
Hey. 

We use `max_margin_contrastive_loss` instead of the supervised NT-Xent loss in our code as it's a bit faster to compute. Regarding your original question, `base_temperature` acts as another scaling parameter to further push the original temperature as per my understanding.  I would also suggest reaching out to Yonglong (yonglong@mit.edu) (who authored the loss) for further details.
